src/main/groovy/io/seqera/wave/controller/RegistryProxyController.groovy

```diff
@@ -129,6 +133,15 @@ class RegistryProxyController {
             rateLimiterService?.acquirePull( new AcquireRequest(route.identity.userId as String, ip) )
         }
 
+        //increment metrics
+        if( httpRequest.method==HttpMethod.GET && (route.manifest || route.isTag()) ) {
+            CompletableFuture.supplyAsync (() -> metricsService.incrementPullsCounter(route.identity.accessToken))
```

Comment:
Let's use the IO executor for this, for [example](https://github.com/seqeralabs/wave/blob/5056a72e80a56e96160af2ff1cb02aba610d59f6/src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy#L72-L74)

Reply:
done